druid-handler/src/java/org/apache/hadoop/hive/ql/io/DruidOutputFormat.java

```diff
@@ -0,0 +1,393 @@
+                                                        .readValue("{\"type\":\"offline\"}");
+      }
+      catch (IOException e) {
+        Throwables.propagate(e);
+      }
+      this.tuningConfig = Preconditions.checkNotNull(realtimeTuningConfig);
+
+
+      this.dataSchema = dataSchema;
+      appenderator = defaultOfflineAppenderatorFactory.build(
+          this.dataSchema,
+          tuningConfig,
+          new FireDepartmentMetrics()
+      );
+      this.maxPartitionSize = maxPartitionSize == null ? DEFAULT_MAX_PARTITION_SIZE : maxPartitionSize;
+
+      appenderator.startJob(); // maybe we need to move this out of the constructor
+    }
+
+    private SegmentIdentifier getSegmentIdentifier(DruidWritable druidHiveRecord)
+    {
+      return getSegmentIdentifier(Longs.tryParse((String) druidHiveRecord.getValue()
+                                                                         .get(DruidTable.DEFAULT_TIMESTAMP_COLUMN)));
+    }
+
+    private SegmentIdentifier getSegmentIdentifier(long timestamp)
+    {
+
+      final Granularity segmentGranularity = dataSchema.getGranularitySpec().getSegmentGranularity();
+
+
+      final long truncatedTime = segmentGranularity.truncate(new DateTime(timestamp)).getMillis();
+
+      List<SegmentIdentifier> segmentIdentifierList = segments.get(truncatedTime);
+
+      final Interval interval = new Interval(
+          new DateTime(truncatedTime),
+          segmentGranularity.increment(new DateTime(truncatedTime))
+      );
+
+      SegmentIdentifier retVal;
+      if (segmentIdentifierList == null) {
+
+        retVal = new SegmentIdentifier(
+            dataSchema.getDataSource(),
+            interval,
+            tuningConfig.getVersioningPolicy().getVersion(interval),
+            new LinearShardSpec(0)
+        );
+        segments.put(truncatedTime, Arrays.asList(retVal));
+        return retVal;
+      } else {
+        retVal = segmentIdentifierList.get(segmentIdentifierList.size());
+        int rowCount = appenderator.getRowCount(retVal);
+        if (rowCount < maxPartitionSize) {
+          return retVal;
+        } else {
+          retVal = new SegmentIdentifier(
+              dataSchema.getDataSource(),
+              interval,
+              tuningConfig.getVersioningPolicy().getVersion(interval),
+              new LinearShardSpec(segmentIdentifierList.size())
+          );
+          segmentIdentifierList.add(retVal);
+          return retVal;
+        }
+      }
+    }
+
+    @Override
+    public void write(Writable w) throws IOException
+    {
+      DruidWritable record = (DruidWritable) w;
+      InputRow inputRow = new MapBasedInputRow(
+          Longs.tryParse((String) record.getValue()
+                                        .get(DruidTable.DEFAULT_TIMESTAMP_COLUMN)),
+          dataSchema.getParser()
+                    .getParseSpec()
+                    .getDimensionsSpec()
+                    .getDimensionNames(),
+          record.getValue()
```

Comment:
how about we don't add it ? does that makes it more complicated for early stages ? just wondering.